### ISSUE TITLE
what a whitespace character is

### ISSUE THREAD

**Issue by r12a** [w3c/i18n-activity#96]
**This is a tracker issue.** Only discuss things here if they are i18n WG internal meta-discussions about the issue. **Contribute to the actual discussion at the following link:**

§ https://github.com/w3c/csswg-drafts/issues/3754

**Comment by KatyMawhood** [comment]
This has been inactive for a couple of years; it feels appropriate for review because there is a continued lack of interop between white spaces. This is indicated by the [test case](https://lists.w3.org/Archives/Public/www-archive/2015Aug/att-0020/first-letter-form-feed.html) using a ::first-letter pseudo-element.

I propose to review David's suggestions:

(a) we should have interop on the ::first-letter whitespace
skipping characters, and it should be specified
(b) ::blank, or whatever we call it, should use the same definition
of whitespace, since I don't want two definitions of "text
that's only whitespace" in selectors

**Comment by r12a** [comment]
Since nothing seems to have happened after Fantasai's proposal, perhaps we should raise an issue for CSS Text asking whether there is a plan to implement that proposal at some point.

**Comment by r12a** [comment]
Useful links:

https://infra.spec.whatwg.org/#ascii-whitespace

https://drafts.csswg.org/css-text/#white-space-processing

https://drafts.csswg.org/css-syntax/#whitespace-diagram

**Comment by r12a** [comment]
Raised a new issue at https://github.com/w3c/csswg-drafts/issues/3754 asking for a status update.

**Comment by r12a** [comment]
This issue was discussed in [a meeting](https://www.w3.org/2019/03/21-i18n-minutes.html#item08).

- `ACTION: Revive issue 96 with css as a github issue [on Richard Ishida - due 2019-03-28].`
<details><summary><i class="differentiate">View the transcript</i></summary>
<a href="https://github.com/w3c/i18n-activity/issues/96">https://github.com/w3c/i18n-activity/issues/96</a><br/>
what a whitespace character is<br/>
<b>richard:</b> changes to css to define whitespace characters<br/>
... no indication of implementation<br/>
... went quiet<br/>
&lt;Katy> <a href="https://lists.w3.org/Archives/Public/www-archive/2015Aug/att-0020/first-letter-form-feed.html">https://lists.w3.org/Archives/Public/www-archive/2015Aug/att-0020/first-letter-form-feed.html</a><br/>
<b>richard:</b> raise as github issue asking "what happened here?"<br/>
<b>&lt;scribe&gt;</b> ACTION: richard: revive issue 96 with CSS as a github issue<br/>
<b>&lt;trackbot&gt;</b> Created ACTION-789 - Revive issue 96 with css as a github issue [on Richard Ishida - due 2019-03-28].<br/>
</details>

**Comment by himorin** [comment]
issue at the working group repository has been closed.

**Comment by himorin** [comment]
removing close? label, original issue reopened.

**Comment by r12a** [comment]
I updated the initial comment to make this a tracker item for the new issue at https://github.com/w3c/csswg-drafts/issues/3754.  (There was no other discussion on this topic after the initial email thread stopped.)

**Comment by r12a** [comment]
I also set the i18n label on the CSS issue to i18n-tracking, rather than i18n-comment.

**Comment by r12a** [comment]
I don't think we need to comment further on this. We're currently only tracking; the CSS issue is open, but not active.  I think we should just observe and wait for something else to happen.

**Comment by xfq** [comment]
Changing the spec label from `s:css-text` to `s:selectors` to match the CSSWG label.

**Comment by xfq** [comment]
CSSWG issue closed. They have determined that `:blank` no longer relies on the definition of "whitespace". And `:empty` continues to rely on the "document white space" defined in CSS Text. As for whether to fully align the whitespace definition in CSS Text with the ASCII whitespace defined in Infra, CSSWG considers this to be a separate compatibility issue that should not be resolved within the scope of this Selectors issue.

Maybe closing our tracker issue?
